gravatar-ui/src/main/java/com/gravatar/ui/components/State.kt

```diff
@@ -14,38 +14,39 @@ import com.gravatar.api.models.Profile
 import com.gravatar.api.models.VerifiedAccount
 import com.gravatar.extensions.emptyProfile
 import com.gravatar.ui.GravatarTheme
-import com.gravatar.ui.components.UserProfileState.Loaded
-import com.gravatar.ui.components.UserProfileState.Loading
+import com.gravatar.ui.components.State.Loaded
+import com.gravatar.ui.components.State.Loading
 import kotlinx.coroutines.delay
 import java.net.URI
 
 /**
- * [UserProfileState] represents the state of a user profile loading.
+ * [State] represents the state of a user profile loading.
  * It can be in a [Loading] state or a [Loaded] state.
  */
-public sealed class UserProfileState {
+public sealed class State<out T> {
```

Comment:
I'm doubting about `State` or `GravatarState`. Wdyt?

Reply:
I'm worried `State` will conflict with user-defined classes. Let's be honest, we all have these generically-named classes in our projects. On the other hand `GravatarState` is misleading because in my mind it would refer to the state of the whole SDK. `GravatarComponentState` maybe?

Reply:
> I'm worried State will conflict with user-defined classes.

With the auto-complete of the modern IDE, I wouldn't say it's a big deal. But yeah, I see the point.

> `GravatarComponentState` maybe?

I'm not 100% done with it, but given that, I don't have a better option 😅 let's change it.

Reply:
Done!